src/client/js/components/toast-alert.js

```diff
@@ -126,33 +126,37 @@ customElements.define('toast-alert', class extends HTMLElement {
   constructor () {
     super()
     this.attachShadow({ mode: 'open' })
```

Comment:
Oh nice, I did not know that that also returned the shadowroot! It works right now because `attachShadow` has the side effect of setting `this.shadowRoot`, but apparently it _also_ returns it.

Love to get me rid of some `@ts-ignore`s: 15fb7eada

Reply:
OK, alternate approach: we just tell TypeScript that `this.shadowRoot` is always set. That means it will no longer catch errors if we forget to call `this.attachShadow({ mode: 'open' })` first, but that looks like a fair trade-off: ccc691793